mmpy_bot/plugins/base.py

```diff
@@ -75,19 +110,6 @@ def on_stop(self):
         log.debug(f"Plugin {self.__class__.__name__} stopped!")
         return self
 
-    async def call_function(
-        self,
-        function: Function,
-        event: EventWrapper,
-        groups: Optional[Sequence[str]] = [],
-    ):
-        if function.is_coroutine:
-            await function(event, *groups)  # type:ignore
-        else:
-            # By default, we use the global threadpool of the driver, but we could use
-            # a plugin-specific thread or process pool if we wanted.
-            self.driver.threadpool.add_task(function, event, *groups)
-
```

Comment:
Can you explain this? This already had access to the driver, so what problem does moving it out of the class solve?

Reply:
Code duplication mostly. At the moment, this code needs to be used in `Plugin` and `PluginManager` due to how `call_function` is triggered in the driver. If we move the listeners out of the event handler class and make `PluginManager` less transparent this may no longer be necessary. I didn't get that far.